### ISSUE TITLE
Remove CACHE_EMBEDDED_REMOTE_STORE Feature

### ISSUE THREAD

**Issue by ryanemerson** [keycloak/keycloak#34160]
### Description

> [!NOTE]
> This is a candidate to move to 26.3 

The intention is to remove `CACHE_EMBEDDED_REMOTE_STORE` in Keycloak 26.2 if users are happy migrating to the new capabilities introduced in 26.

### Details

- The `spi-connections-infinispan-quarkus-site-name` configuration property can be removed as it is only required by `CACHE_EMBEDDED_REMOTE_STORE`

**Comment by ahus1** [comment]
As noted by @pedroigor in https://github.com/keycloak/keycloak/pull/33297#issuecomment-2429276162 this option is no longer needed for the KC26 style multisite setup. 

Still we might need it temporarily for a fallback option make multi-site JDBC Ping work as we kept it in the docs, see https://github.com/keycloak/keycloak/pull/33297#issuecomment-2429305357

So a way forward could be: 

* Remove the option for multi-site docs
* Introduce a proper new option for multi-site JDBC, with a fallback to the option we had in KC 26.0 with a warning
* Remove the fallback for KC27.0

**Comment by ahus1** [comment]
Given the latest discussion in #33297, it seems this option can be removed as it is no longer used and can be removed.

**Comment by ryanemerson** [comment]
The `site-name` is a dependency of the `CACHE_EMBEDDED_REMOTE_STORE` Feature. This experimental feature was added to allow Keycloak 24 users to revert to the old multi-site behaviour if the new behaviour of the MULTI_SITE Feature in Keycloak 26 was not acceptable. The intention is to wait for user feedback on the new behaviour and if it is acceptable, ultimately remove the `CACHE_EMBEDDED_REMOTE_STORE` and associated code. Unfortunately, this means we cannot remove the site-name configuration until this Feature has been removed.

**Comment by ahus1** [comment]
Yes, let's use this one for now until we see the need to split it.
